File: package/go/src/age_statistics.go
package src

import (
	"math"
)

type AgeStatistics struct {
	MaxAge         int
	MinAge         int
	AvgAge         float64
	CountBelowAvg  int
}

func AnalyzeAgeDistribution(ages []int) AgeStatistics {
	if len(ages) == 0 {
		return AgeStatistics{}
	}

	maxAge := ages[0]
	minAge := ages[0]
	sum := 0

	for _, age := range ages {
		if age > maxAge {
			maxAge = age
		}
		if age < minAge {
			minAge = age
		}
		sum += age
	}

	avgAge := math.Round(float64(sum)/float64(len(ages))*100) / 100

	countBelowAvg := 0
	for _, age := range ages {
		if float64(age) <= avgAge {
			countBelowAvg++
		}
	}

	return AgeStatistics{
		MaxAge:        maxAge,
		MinAge:        minAge,
		AvgAge:        avgAge,
		CountBelowAvg: countBelowAvg,
	}
}

File: package/typescript/src/ageStatistics.ts
export function analyzeAgeDistribution(ages: number[]): [number, number, number, number] {
  if (ages.length === 0) {
    throw new Error('Ages array cannot be empty');
  }

  const maxAge = Math.max(...ages);
  const minAge = Math.min(...ages);
  const avgAge = Math.round((ages.reduce((sum, age) => sum + age, 0) / ages.length) * 100) / 100;
  const countBelowAvg = ages.filter(age => age <= avgAge).length;

  return [maxAge, minAge, avgAge, countBelowAvg];
}
